app/src/evaluation/metrics/runner.py

```diff
@@ -2,14 +2,38 @@
 
 import csv
 import os
-from typing import Any, Dict, List, Optional
+from typing import Any, Callable, Dict, List, Optional, Sequence
+
+from src.retrieve import retrieve_with_scores
 
 from .batch import create_batch_config, filter_questions, stratified_sample
 from .logging import EvaluationLogger
 from .metric_computation import compute_metrics_summary
 from .results import batch_process_results
 
 
+def create_retrieval_function(
+    min_score: Optional[float] = None,
+) -> Callable[[str, int], Sequence[Any]]:
+    """Create a function to retrieve chunks for a question.
+
+    Args:
+        min_score: Optional minimum similarity score for retrieval
+
+    Returns:
+        Function that takes a question and k value and returns retrieved chunks
+    """
+
+    def retrieval_func(query: str, k: int) -> Sequence[Any]:
+        # Default to -1.0 if no min_score provided
+        score_threshold = min_score if min_score is not None else -1.0
+        return retrieve_with_scores(
+            query=query, retrieval_k=k, retrieval_k_min_score=score_threshold
+        )
+
+    return retrieval_func
```

Comment:
Consider using `functools.partial here`:
```suggestion
    from functools import partial
    score_threshold = min_score if min_score is not None else -1.0
    return partial(retrieve_with_scores, retrieval_k_min_score=score_threshold)
```

Reply:
Didn't end up including this suggestion since `results.batch_process_results` uses it as `func(query, k)`. But I agree your solution is more concise. I will look into applying this at a later time.

Reply:
> `results.batch_process_results` uses it as `func(query, k)`

That should still work, no?